main.py

```diff
@@ -45,77 +49,100 @@ def get_eurojackpot_results():
+    return [EuroJackpot(e) for e in r]
 
 
-def generate_discord_msg():
-    primary_result, secondary_result, prize_tiers = get_eurojackpot_results()
-    if not primary_result:
-        return "Tuloksia ei saatu Veikkaukselta :("
+def fetch_winnings(
+        eurojackpot: EuroJackpot,
+        guesses_primary: List[str],
+        guesses_secondary: List[str]) -> Tuple[int, int, int, int]:
+    primary_results = eurojackpot.results[0].primary
+    secondary_results = eurojackpot.results[0].secondary
 
     primary_hits = 0
-    for number in primary_numbers:
-        if number in primary_result:
+    for number in primary_results:
+        if number in guesses_primary:
             primary_hits += 1
 
     secondary_hits = 0
-    for number in secondary_numbers:
-        if number in secondary_result:
+    for number in secondary_results:
+        if number in guesses_secondary:
             secondary_hits += 1
 
-    result = f"{primary_hits}+{secondary_hits} oikein"
+    total_hits = f"{primary_hits}+{secondary_hits} oikein"
 
     money_won = 0
-
-    for tier in prize_tiers:
-        if tier["name"] == result:
-            money_won = int(tier["shareAmount"])
+    for prize_tier in eurojackpot.prize_tiers:
+        if prize_tier.name == total_hits:
+            money_won = prize_tier.share_amount
+            break
 
     investment_value_old = get_investment_value()
-    investment_value = str(investment_value_old - 200 + money_won)
+    investment_value_new = investment_value_old - 200 + money_won
+
+    set_investment_value(investment_value_new)
+
+    return primary_hits, secondary_hits, money_won, investment_value_new
+
 
-    set_investment_value(investment_value)
+def generate_discord_msg(guesses_primary: List[Union[int, str]], guesses_secondary: List[Union[int, str]]) -> str:
+    guesses_primary = [str(i) for i in guesses_primary]
+    guesses_secondary = [str(i) for i in guesses_secondary]
 
-    return f"<@&{discord_group_id}> {result}, voittoa {int(money_won)/100:.2f}€, sijoituksen tuotto ||{int(investment_value)/100:.2f}€||"
+    ejackpot_messages = []
+    results = get_eurojackpot_results()
+    if not results:
+        return "Tuloksia ei saatu Veikkaukselta :("
+
+    for result in results:
+        info = fetch_winnings(result, guesses_primary, guesses_secondary)
+
+        primary_hits = info[0]
+        secondary_hits = info[1]
+        money_won = info[2]
+        investment_value = info[3]
+
+        ejackpot_week = datetime.datetime.fromtimestamp(result.close_time/1000).isocalendar().week
+        game = result.brand_name.split("-")[1]
+        weekday = result.brand_name.split("-")[0]
+        msg = f"{game} viikko {ejackpot_week}/{weekday}, " \
+              f"{primary_hits}+{secondary_hits} oikein, " \
+              f"voittoa {int(money_won)/100:.2f}€, " \
+              f"sijoituksen tuotto ||{investment_value/100:.2f}€||"
+        ejackpot_messages.append(msg)
```

Comment:
I have a trigger that runs this bot at Tuesdays and Fridays, so this would be duplicate information when it is running twice a week. This is useful information, but I would prefer to get only the latest result of the week

Reply:
One option could be to make the `get_results` function take some boolean like `latest_only`:

```python
def get_eurojackpot_results(latest_only: bool) -> List[EuroJackpot]:
    now = datetime.datetime.now()

    week = now.isocalendar().week
    year = now.isocalendar().year

    r = requests.get(
        f"https://www.veikkaus.fi/api/draw-results/v1/games/EJACKPOT/draws/by-week/{year}-W{week}").json()

    if latest_only:
        return [EuroJackpot(r[-1])]
    else:
        return [EuroJackpot(e) for e in r]
```

The boolean could then be parsed from the env file, and have a default value of True so that only the latest results are returned for message generation, but both options still being available from the env file.

Opinions on this?

Reply:
I think `latest_only` flag from env variable would be ideal solution here 👍

Reply:
I actually read the new code today but didn't have time to resolve the conflicts yet. I'll merge the mainbranch  into this branch soon 👍